server/py/framework/db/sqldb/db.py

```diff
@@ -5887,6 +5888,83 @@ def _delete_alert_notifications(
         if commit:
             session.commit()
 
+    def store_alert_activation(
+        self,
+        session,
+        alert_data: mlrun.common.schemas.AlertConfig,
+        event_data: mlrun.common.schemas.Event,
+        notifications_states: list[mlrun.common.schemas.NotificationState],
+    ):
+        extra_data = {
+            "notifications": [
+                notification.dict() for notification in notifications_states
+            ],
+            "criteria": alert_data.criteria.dict() if alert_data.criteria else None,
+        }
+
+        # For JOB entities, construct entity_id as "name.uid" format
+        if alert_data.entities.kind == mlrun.common.schemas.alert.EventEntityKind.JOB:
+            run_name = alert_data.entities.ids[0]
+            run_uid = event_data.value_dict.get("uid")
+            entity_id = f"{run_name}.{run_uid}"
```

Comment:
Not sure if it is even possible to not have a `uid` in the event for a JOB failure, but better be defensive and not end up with random periods.
```suggestion
            run_uid = event_data.value_dict.get("uid")
            entity_id = f"{run_name}.{run_uid}" if run_uid else run_name
```